src/Poster.js

```diff
@@ -1,17 +1,61 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './Poster.css';
+import { movieDetailsById } from './apiCalls.js';
 import { Link } from 'react-router-dom';
 
 const Poster = ({image, id, isSinglePoster}) => {
+    // Updateing app state here causes an infinite re-render
```

Comment:
Love the sneak peek on using hooks here. Also, knowing that there is more than one way to solve this problem is great to know.